docs/Design/Null Checking.md

```diff
@@ -0,0 +1,148 @@
+# Null Checking
+
+Creates checks for if the passed variable has value null. 
+
+A null variable is one that has no other value stored in it.
+
+Can be used to check whether the value is or is not null. 
+
+## Commands
+
+### `is null`
+
+`is null : variable`
+
+Creates a test that will return true if the variable is `null`.
+
+### `not null`
+
+`not null : variable`
+
+Creates a test that will return false if the variable is `null`.
+
+## Usage
+
+```
+is null : foo
+```
+
+```
+not null : bar
+```
+
+### CSharp
+
+```csharp
+foo == null
+```
+
+```csharp
+bar != null
+```
+
+### Java
+
+```java
+foo == null
+```
+
+```java
+bar != null
+```
+
+### Python
+
+```python
+foo is None
+```
+
+```python
+bar is not None
+```
+
+### Ruby
+
+```ruby
+foo.nil?
+```
+
+```ruby
+!bar.nil?
+```
+
+### TypeScript
+
+```typescript
+typeof foo === undefined
+```
+
+```typescript
+typeof foo !== undefined
+```
+
+## Implementation
+
+A null check consists of a variable to be checked, and some logical equivalence operator.
+
+### Properties

```

Comment:
These properties should be (practically/basically) **everything** GLS needs to compute the operator for the language. What you have now isn't quite enough.

The properties alone should be enough to inform GLS whether `is null : foo` should be `foo != null` or `typeof foo !== undefined` (but the correct TypeScript way :) ) or etc.